Tryouts/Core/BasicModels/Modules/ProcessDockLayoutBehavior.cs

```diff
@@ -0,0 +1,234 @@
+/// See the License for the specific language governing permissions and limitations under the License.
+/// 
+/// ********************************************************************************************************
+
+using NP.Avalonia.UniDockService;
+using NP.Concepts.Behaviors;
+using System;
+using System.Reactive.Linq;
+
+namespace MorganStanley.ComposeUI.Tryouts.Core.BasicModels.Modules
+{
+    public class ProcessDockLayoutBehavior
+    {
+        IUniDockService _uniDockService;
+        ProcessesViewModel _processesViewModel;
+        string _dockSerializationFileName;
+        string _vmSerializationFileName;
+        string _mainGroupDockId;
+        string _contentTemplateResourceKey;
+        string _headerContentTemplateResourceKey;
+
+        int _newDockId = 0;
+
+        SynchronizationContext _synchronizationContext;
+
+        IDisposable? _uniDockVmBehavior;
+        IDisposable? _processesWithWindowsBehavior;
+
+        public ProcessDockLayoutBehavior
+        (
+            IUniDockService uniDockService,
+            ProcessesViewModel processesViewModel,
+            string dockSerializationFileName,
+            string vmSerializationFileName,
+            string mainGroupDockId,
+            string contentTemplateResourceKey,
+            string headerContentTemplateResourceKey
+        )
+        {
+            _uniDockService = uniDockService;
+            _processesViewModel = processesViewModel;
+            _dockSerializationFileName = dockSerializationFileName;
+            _vmSerializationFileName = vmSerializationFileName;
+            _mainGroupDockId = mainGroupDockId;
+            _contentTemplateResourceKey = contentTemplateResourceKey;
+            _headerContentTemplateResourceKey = headerContentTemplateResourceKey;
+
+#pragma warning disable CS8974 // Converting method group to non-delegate type
+            _processesWithWindowsBehavior = 
+                _processesViewModel.ProcessesWithWindows.AddBehavior(OnProcessWithWindowAdded, OnProcessWithWindowRemoved);
+#pragma warning restore CS8974 // Converting method group to non-delegate type
+
+            _synchronizationContext = SynchronizationContext.Current!;
+
+            ResetBehavior();
+        }
+
+        private void ResetBehavior()
+        {
+            _uniDockVmBehavior?.Dispose();
+
+            _uniDockVmBehavior = _uniDockService.DockItemsViewModels.AddBehavior(OnVmAdded, OnVmRemoved);
+        }
+
+        private void OnVmAdded(DockItemViewModelBase vm)
+        {
+
+        }
+
+        private void OnVmRemoved(DockItemViewModelBase vm)
+        {
+            if (_isLoading)
+            {
+                return;
+            }
+            DockItemViewModel<ProcessData> processVm = 
+                (DockItemViewModel<ProcessData>)vm;
+
+            Guid instanceId = 
+                processVm.TheVM!.InstanceId;
```

Comment:
What's "TheVM"?

Reply:
this is the actual View Model of type ProcessData. This is part of the UniDock framework.